pkg/v2/tkr/util/topology/cluster.go

```diff
@@ -154,3 +162,38 @@ func removeMDVariableForName(md *clusterv1.MachineDeploymentTopology, name strin
 		}
 	}
 }
+
+// GetClusterAvailableUpgrades returns the available upgrades of a cluster
+func GetClusterAvailableUpgrades(cluster *clusterv1.Cluster) (sets.StringSet, error) {
+	updatesMsg := ""
+	for _, condition := range cluster.Status.Conditions {
+		if condition.Type == runv1alpha3.ConditionUpdatesAvailable && condition.Status == corev1.ConditionTrue {
+			updatesMsg = condition.Message
+			break
+		}
+	}
+	updates := updatesFromConditionMessage(updatesMsg)
+	if len(updates) == 0 {
+		return sets.Strings(), tkrutils.NewUpgradesNotAvailableError(fmt.Sprintf("no available upgrades for cluster '%s', namespace '%s'", cluster.Name, cluster.Namespace))
+	}
+	return updates, nil
+}
+
+func updatesFromConditionMessage(updatesMsg string) sets.StringSet {
+	if updatesMsg == "" {
+		return sets.Strings()
+	}
+	// Example for message - [<tkr-version-1> <tkr-version-2>]"
+	updates := strings.Split(strings.TrimRight(strings.TrimLeft(updatesMsg, "["), "]"), " ")
+	if len(updates) == 0 {
+		return sets.Strings()
+	}
+	updatesSet := sets.Strings(updates...)
+
+	return updatesSet.Map(func(tv string) string {
+		if !strings.HasPrefix(tv, "v") {
+			tv = "v" + tv
+		}
+		return tv
+	})
```

Comment:
No need to check an add the `v` prefix: all these versions should already have it as they are coming from TKRs / Cluster UpdatesAvailable condition.

Reply:
Updated. Thanks!